js/slider.js

```diff
@@ -0,0 +1,21 @@
+
+var slideIndex = 0;
+showSlides();
+
+function showSlides() {
+  var i;
+  var slides = document.getElementsByClassName("mySlides");
+
+  for (i = 0; i < slides.length; i++) {
```

Comment:
This loop can also be moved outside the function. We don't need to hide every slide every time. We just have to hide the current shown slide and make next slide visible.